### PR TITLE
[Token] Add update collection data method

### PR DESCRIPTION
**Pull request by ericEnjoy**
### Description
We need to update collection data such as description or collection uri currently, found the data structrue already preserved but no method found. So add these three entry functions.

### Test Plan

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/4103)

### CODE REVIEW

File: aptos-move/framework/aptos-token/sources/token.move [hunk 1]
```diff
@@ -300,6 +300,64 @@ module aptos_token::token {
...
+        assert!(exists<Collections>(creator), ECOLLECTIONS_NOT_PUBLISHED);
+        let collections = borrow_global_mut<Collections>(
+            creator
+        );
+        let collection_data = check_collection_return_data(collections, collection_name);
+        assert!(collection_data.mutability_config.maximum, EFIELD_NOT_MUTABLE);
+        collection_data.maximum = maximum;
+    }
+
+    fun check_collection_return_data(
```
Comment (davidiw) [inline aptos-move/framework/aptos-token/sources/token.move:354]: why not borrow_collection_data_mut

**Review by davidiw — APPROVED** [review]

File: aptos-move/framework/aptos-token/sources/token.move (hunk 1 above)
Comment (ericEnjoy) [reply aptos-move/framework/aptos-token/sources/token.move:354]: Seems a better name, I will update it

File: aptos-move/framework/aptos-token/sources/token.move [hunk 2]
```diff
@@ -1149,4 +1207,21 @@ module aptos_token::token {
...
+        // token owner mutate the token property
+        let token_id = create_collection_and_token(creator, 2, 4, 4);
+        assert!(token_id.property_version == 0, 1);
+
+        mutate_collection_description(
+            creator,
+            token_id.token_data_id.creator,
+            token_id.token_data_id.collection,
+            string::utf8(b"new_description")
+        );
```
Comment (areshand) [inline aptos-move/framework/aptos-token/sources/token.move:1225]: could you add assert to show that the new description is updated to new_description?

Comment (ericEnjoy) [reply aptos-move/framework/aptos-token/sources/token.move:1252]: yes, just added it.

**Pull request closed without merging**